libs/repositories/src/UsdRepository/UsdRepositoryCoingecko.ts

```diff
@@ -49,20 +48,23 @@ export class UsdRepositoryCoingecko implements UsdRepository {
       }
     );
 
-    if (fetchResponse.error) {
-      throw fetchResponse.error;
-    }
-    const priceData = fetchResponse.data;
-
-    if (!priceData[tokenAddressLower] || !priceData[tokenAddressLower].usd) {
+    if (
+      response.status === 404 ||
```

Comment:
What's the behaviour when it throws? 500 or return the same 4xx error?

Reply:
The behaviour of 500 for me is similar to the any unhandled 400, I just want to throw, because is not the 200 I'm after (or the other 2 cases where I return a `null` stating I don't know the price)

In other words. I handle all expected cases, the unexpected, they throw

Reply:
It's still not clear to me.
Would a 400 response from coingecko result in a 500 response from bff?

Reply:
> Would a 400 response from coingecko result in a 500 response from bff?

It should, yes

Reply:
A little, yes.
Image we have an alert for high rate of 500s that will wake someone up to deal with it.
The errors though are coming from a client (potentially not us) that's simply using the wrong parameters, or we ourselves called the upstream endpoint too often, causing a 429.

Should the service still return 500s? What would be the action oncall can take to alleviate the 500s?

I raise this mostly because 5xx errors are in my mind something unexpected that should not be happening such as a server crash. A 4xx error from an upstream service is just expected and the service is still running fine.

Reply:
I'll open an issue so we can keep track of this, cause I think this needs some alignment, although I see improvements on capturing more cases from falling upstream, I do think that us passing bad params to an API we delegate is an internal error on our side, not something the client did wrong.

I do think this should trigger this 500 alert in case we messed up the params, or the underling delegation service changed the params. 40X would mean its the user's fault, but 50X means its our fault. 

Then, its possible we would need to refine further and catch some cases and handle it differently. We might want to decide what to do in case of a 429. Again, this is not necessarily users's fault (it could be we have multiple concurrent users and this triggers the error, so again the issue is in our side, nothing the client can fix). The refinement could be to add some cool down and re-attempt, throttle requests so this doesn't happen, or pay for a better API Key that allows more request/second. 

Anyways, let's discuss it in an issue or personally to get an alignment

Reply:
https://github.com/cowprotocol/bff/issues/69

Reply:
Ok, fair points